transmart-rest-api-e2e/src/test/groovy/tests/rest/v2/CrossTableSpec.groovy

```diff
@@ -3,99 +3,103 @@ package tests.rest.v2
+        request.user = DEFAULT_USER
+        request.statusCode = HttpStatus.FORBIDDEN.value()
+        def errorResponse = post(request) as ErrorResponse
 
-        then: "Returned counts for all cells equal zero"
-        assert responseData2.rows.size() == 2
-        assert responseData2.rows[0] == [0, 0]
-        assert responseData2.rows[1] == [0, 0]
+        then: 'Access is denied'
+        errorResponse.httpStatus == HttpStatus.FORBIDDEN.value()
+    }
 
-        when: "I do not have an access to the restricted concept path constraint"
-        def patientSetRequest2 = [
-                path      : PATH_PATIENT_SET,
+    @RequiresStudy([EHR_ID, SHARED_CONCEPTS_RESTRICTED_ID, SHARED_CONCEPTS_A_ID])
+    def "access denied for cross table for restricted constraints"() {
+        given: 'I do not have an access to the restricted concept path constraint'
+        def restrictedConceptCode = 'SCSCP:DEM:AGE'
+        def params = [
+                rowConstraints   : [[type: TrueConstraint],
+                                    [type: StudyNameConstraint, studyId: EHR_ID]],
+                columnConstraints: [[type: ConceptConstraint, conceptCode: restrictedConceptCode],
+                                    [type: TrueConstraint]],
+                subjectConstraint: [type: TrueConstraint],
+        ]
+        def request = [
+                path      : PATH_CROSSTABLE,
                 acceptType: JSON,
-                query     : [name: 'crosstable_test_set'],
-                body      : [
-                        type    : Combination,
-                        operator: OR,
-                        args    : [
-                                [type: StudyNameConstraint, studyId: EHR_ID],
-                                [type: StudyNameConstraint, studyId: SHARED_CONCEPTS_A_ID]
-                        ]
-                ],
-                user      : ADMIN_USER,
-                statusCode: 201
+                user      : DEFAULT_USER,
+                body      : params,
+                statusCode: HttpStatus.FORBIDDEN.value()
         ]
-        patientSetRequest2.user = DEFAULT_USER
-        def patientSetResponse2 = post(patientSetRequest2) as Map
-        def patientSetId2 = patientSetResponse2.id
-        def subjectConstraint2 = [type: PatientSetConstraint, patientSetId: patientSetId2]
-        def request3 = request2
-        request3.body.subjectConstraint = subjectConstraint2
-        def responseData3 = post(request3) as Map
 
-        then: "Returned count for the first cell equals zero"
-        assert responseData3.rows.size() == 2
-        assert responseData3.rows[0] == [0, 5]
-        assert responseData3.rows[1] == [0, 3]
+        when: 'I request a cross table with a reference to a restricted concept in one of the constraints'
+        def errorResponse = post(request) as ErrorResponse
+
+        then: 'Access is denied'
+        errorResponse.httpStatus == HttpStatus.FORBIDDEN.value()
```

Comment:
Nice! I have another test in transmart-rest-api func. tests to remove `@Ignore` after the change.